ansible_collections/serverscom/sc_api/tests/integration/targets/sc_baremetal_os_list/tasks/main.yaml

```diff
@@ -0,0 +1,75 @@
+---
+- name: Check if there are sc_token and sc_endpoint variables
+  no_log: true
+  fail:
+    msg: "You need to define sc_token and sc_endpoint variables in tests/integration/integration_config.yml"
+  when: not sc_endpoint or not sc_token
+
+- name: Test invalid token
+  sc_baremetal_os_list:
+    token: invalid
+    server_id: "{{ existing_server1_id}}"
+  register: test1
+  failed_when:
+    - test1 is success
+    - test1.status_code != 401
+
+- name: Test2, Get OS list for a specific server
+  sc_baremetal_os_list:
+    token: "{{ sc_token }}"
+    server_id: "{{ existing_server1_id}}"
+  register: test2
+
+- name: Check Test2
+  assert:
+    that:
+      - test2 is not changed
+      - test2.os_list | length > 0
+      - test2.os_list[0].name
+      - test2.os_list[0].full_name
+      - test2.os_list[0].version
+      - test2.os_list[0].arch
+      - test2.os_list[0].id
+      - test2.os_list[0].filesystems | length > 0
+
+- name: Test3, Get OS list by name regex
```

Comment:
Please, add negative tests (no OS found). The failure path is rather complicated, so it should be tested too.